File: java/src/main/java/wot/tak/connection/UDPConnector.java
package wot.tak.connection;

import java.io.BufferedReader;
import java.io.File;
import java.io.FileReader;
import java.io.IOException;
import java.net.DatagramPacket;
import java.net.DatagramSocket;
import java.net.InetAddress;

public class UDPConnector implements TAKServerConnector {
    public final String url;
    public final int port;

    private DatagramSocket socket;
    private InetAddress address;

    public UDPConnector(String url, String port) {
        this.url = url;
        this.port = Integer.parseInt(port);
    }

    @Override
    public void connect() throws IOException {
        socket = new DatagramSocket(this.port);
        address = InetAddress.getByName(url);
    }

    @Override
    public void sendFile(File file) throws IOException {
        StringBuilder msg = new StringBuilder();
        try (BufferedReader br = new BufferedReader(new FileReader(file))) {
            String line;
            while ((line = br.readLine()) != null) {
                msg.append(line);
            }
        }
        byte[] data = msg.toString().getBytes();
        socket.send(new DatagramPacket(data, data.length, address, port));
    }

    @Override
    public void close(){
        socket.close();
    }
}

File: c#/src/WOT.TAK.Connection/src/UDPConnector.cs
using System.Net.Sockets;
using System.Text;

namespace WOT.TAK.Connection
{
    public class UDPConnector : TAKServerConnector
    {
        private readonly string _url;
        private readonly int _port;

        private UdpClient _socket;

        public UDPConnector(string url, string port)
        {
            _url = url;
            _port = int.Parse(port);
            _socket = new UdpClient();
        }

        public void Connect()
        {
            try
            {
                _socket.Connect(_url, _port);
            }
            catch (Exception e ) 
            {
                Console.WriteLine(e.Message);
            }
        }   

        public void SendFile(string filename)
        {
            StringBuilder msg = new StringBuilder();
            using (StreamReader file = new(filename))
            {
                string line;
                while ((line = file.ReadLine()) != null)
                {
                    msg.Append(line);
                }
                file.Close();
            }
            byte[] data = Encoding.ASCII.GetBytes(msg.ToString());
            _socket.Send(data, data.Length);
        }
        public void Close()
        {
            _socket.Close();
        }
    }
}
